src/operator/takeWhile.ts

```diff
@@ -27,11 +25,15 @@ class TakeWhileSubscriber<T> extends Subscriber<T> {
 
   protected _next(value: T): void {
     const destination = this.destination;
-    const result = tryCatch(this.predicate)(value, this.index++);
+    let result: any;
+    try {
+      result = this.predicate(value, this.index++);
+    } catch (err) {
+      destination.error(err);
+      return;
+    }
 
-    if (result == errorObject) {
-      destination.error(errorObject.e);
-    } else if (Boolean(result)) {
+    if (Boolean(result)) {
```

Comment:
Try moving this to it's own function so it's not in the deoptimized function with the try/catch.